include/transmem/transmem.h

```diff
@@ -59,41 +60,64 @@ class EmptyLinkQueryException : public std::runtime_error {
     Link emptyLink;
 };
 
-
-/****************************
- * PATH                     *
- ****************************/
+/********
+ * PATH *
+ ********/
 
 struct Path {
+
     FrameID src;
     FrameID dst;
     std::vector< std::reference_wrapper<Link> > links;
 
     void writeJSON(QJsonObject &json) const;
 };
 
+/************************************
+ * STAMPED AND RATED TRANSFORAMTION *
+ * **********************************/
+
+/* REMARK:
+   New return type of transmem. Encodes not just the transformation but also
+   some information about the quality. Unnecessary convertion between matrix and quaternion
+   are no longer needed and the return type is much more flexible, one can add additional
+   stuff without changing interface to much. */
+
+struct StampedAndRatedTransformation {
+
+    // Quaternions encoding the transformation.
+    QQuaternion qRot;
+    QQuaternion qTra{0,0,0,0};
```

Comment:
I internally use quaternions to represent rotation as well as translation to simplify the combining of multiple transformation along a path. In my eyes its a minor drawback to have the translation part also as a quaternion instead of a vector. I actually read some theory where this representation was used, but i can change that if you don't agree?